File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    ranks = {v: i for i, v in enumerate(players)}
    
    for call in callings:
        i = ranks[call]
        ranks[call] -= 1
        ranks[players[i - 1]] += 1
        players[i - 1], players[i] = players[i], players[i - 1]
        
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.*;

class Solution {
    
    public String[] solution(String[] players, String[] callings) {
        Map<String, Integer> ranks = new HashMap<>();
        
        for (int i = 0; i < players.length; i++) {
            ranks.put(players[i], i);
        }
        
        for (String call : callings) {
            int i = ranks.get(call);
            
            ranks.put(call, i - 1);
            ranks.put(players[i - 1], i);
            
            String temp = players[i - 1];
            players[i - 1] = players[i];
            players[i] = temp;
        }
        
        return players;
    }
}